### DISCUSSION TITLE
Login button in React Test deployment is not responsive, no redirect occur, no error shown.

### DISCUSSION THREAD

**Discussion by dominikusbrian** [authgear/authgear-server#3507] [Get Help] [answered]
Hi there I am facing an issue with unresponsive login button. Which I expect to send me to the portal app after being clicked. 

I checked the code I typed following the React single page app Tutorial, and also tested using the app I copy directly from github authgear-example-react repository without modification. Both exhibit similar behaviour. 

noticing there is no error anywhere could  be found, I am not sure where to check for issue. 

I am using npm 8.19.2 and node v18.12.1 just in case there's compatibility issue I am not aware of. 

Below is a screenshot of the deployed test app. 
![image](https://github.com/authgear/authgear-server/assets/60097047/7d8c0bfe-1db2-48bc-94e0-14e98e6fbed2)

The screenshot for the onClick login button is attached below
![image](https://github.com/authgear/authgear-server/assets/60097047/ebb035f5-5ce1-4e48-86bd-c5c11a8c79fa)

Start login function is as follow:
![image](https://github.com/authgear/authgear-server/assets/60097047/0bf63b68-1ad4-4d2e-942b-798c5feed4a7)

**Comment by chpapa** [comment]
@dominikusbrian any error or messages shown in your Javascript Console? (On Chrome, click menu "View" -> "Developer" -> "Javascript Console"

> also tested using the app I copy directly from github authgear-example-react repository without modification

Do you meant this repo? 

https://github.com/authgear/authgear-example-react

**Reply by dominikusbrian** [reply]
Yes I am testing both with my own version and the github repo you mentioned. 

The following below are inspection for my own version following the [Authgear official React tutorial ](https://docs.authgear.com/get-started/single-page-app/react)

Okay I followed the debug first step you suggested. 
The issue is with CORS setting, I either need to allow it for authgear from my server, or maybe if it is safe enough to specify no-cors instead as suggested in the error code (attached below).

![image](https://github.com/authgear/authgear-server/assets/60097047/4365d56d-91f6-41c2-a130-f3c86465355d)

Meanwhile, I will try to see how I can fix this. Looking forward to great suggestion from the authgear team. 

Best, Domi.

**Reply by dominikusbrian** [reply]
Added the following hook and headers,
![image](https://github.com/authgear/authgear-server/assets/60097047/6a53d300-fc2f-4144-81e0-2974db8bda37)

now there's no error when Login button is clicked 
![image](https://github.com/authgear/authgear-server/assets/60097047/a88df1ec-af6f-471e-87a2-72adcbbfcdca)


but still its not sending me to the authentication page connected to my authgear app [dreambrook.authgear.cloud](https://dreambrook.authgear.cloud)

**Reply by dominikusbrian** [reply]
For the official unabriged copy (deployed in port 4001) from the repository the error are as follows:

![image](https://github.com/authgear/authgear-server/assets/60097047/4b79c3d4-7b9e-41b1-9a5d-0092d13b678f)

To be sure, for deployment with the official repo, I also modified the endpoint app URIs as follow: 

![image](https://github.com/authgear/authgear-server/assets/60097047/b0676e27-4024-435d-92c2-85628d1d632b)

cmiiw~

**Reply by chpapa** [reply, marked as answer]
@dominikusbrian you should not need the hook to work around the CORS header.

When you add the `Authorized Redirect URIs` to Authgear Portal, it shall automatically add the URL to the CORS allow list of Authgear.

Please note that these URL must be exactly the same:

1. The URL on the browser location bar you saw.
2. The `Authorized Redirect URIs`
3. The `redirectURI` in your code.

If the URL is different, e.g. `http://127.0.0.1` vs `http://localhost`, even both URL is the same, due to how browser security works, it would still be invalid.

I saw that in your screenshots you replace something with `localhost`, if you only use `localhost:4000` as URL of all 3 above, would it works? If yes it is probably due to the misconfiguration as mention above.

If it works, would you mind to record the session in HAR file (ref: https://support.zendesk.com/hc/en-us/articles/4408828867098-Generating-a-HAR-file-for-troubleshooting-) and send it to bencheng@authgear.com so we can looks into details?

**Reply by dominikusbrian** [reply]
Hi Ben, thank you for the instructions. 
Unfortunately, without the hook and the work around it will still giving me error regarding CORS header. 

I have changed the URL for point number 2 [The Authorized Redirect URIs] and 3 [The redirectURI in my code] to consistently be http://localhost and not http://127.0.0.1 or explicit IP address.

The one on my browser however is not localhost, because I am deploying the authgear app on an external server and do not do port forwarding for this purpose. So on my browser I am trying to access the page via http://[my server IP address]:4000.  Would this be an issue with browser security ? 

Let me also try deploying locally or via port forwarding to see if that's the case. 

[after trying the above] 
Using port forwarding through VSCode, it works properly now. 
I can type localhost:4000 and now it shows what is expected. Hurray~ 


![image](https://github.com/authgear/authgear-server/assets/60097047/917c8388-c5e7-4d57-9c4d-d8b1ca3de04e)

Thanks for the explanations that help finding where the issue is. 
Next, step is for me to test this on how I can deploy them from the server side without needing to type http://localhost on my user's browser point of view. 

Sure I will try to look into the session recording, I am using Brave browser which have a recording function for user experience, let me see whether it is compatible with the HAR file and will send it to [bencheng@authgear.com](mailto:bencheng@authgear.com) when its done.
